bundler/lib/dependabot/bundler/package_manager.rb

```diff
@@ -23,28 +24,13 @@ class PackageManager < PackageManagerBase
 
       sig { params(version: T.any(String, Dependabot::Version)).void }
       def initialize(version)
-        @version = T.let(Version.new(version), Dependabot::Version)
-        @name = T.let(PACKAGE_MANAGER, String)
-        @deprecated_versions = T.let(DEPRECATED_BUNDLER_VERSIONS, T::Array[Dependabot::Version])
-        @supported_versions = T.let(SUPPORTED_BUNDLER_VERSIONS, T::Array[Dependabot::Version])
-      end
-
-      sig { override.returns(String) }
-      attr_reader :name
-
-      sig { override.returns(Dependabot::Version) }
-      attr_reader :version
-
-      sig { override.returns(T::Array[Dependabot::Version]) }
-      attr_reader :deprecated_versions
-
-      sig { override.returns(T::Array[Dependabot::Version]) }
-      attr_reader :supported_versions
-
-      sig { override.returns(T::Boolean) }
-      def unsupported?
-        # Check if the version is not supported
-        supported_versions.all? { |supported| supported > version }
+        super(
```

Comment:
Moved all common fields name, version, deprecation_versions, supported_version into abstract package manager. 
Also moved `unsupported?` function into abstract package manager since bundler and composer both implements unsupported in the same way.